Source/WebCore/PAL/pal/PALSwift/UnsafeOverlays.swift

```diff
@@ -37,5 +37,67 @@ extension HashFunction {
                 start: data.__dataUnsafe(), count: data.size()))
     }
 }
+extension CryptoKit.Digest {
+    public func toVectorUInt8<T: HashFunction>(_: T.Type) -> VectorUInt8 {
+        var rv = VectorUInt8(T.Digest.byteCount)
+        var index = 0
+        self.forEach { val in
+            rv[index] = val
+            index += 1
+        }
+        return rv
+    }
+
+}
+
+extension Data {
+    public static func fromSpan(spanNoCopy: SpanConstUInt8) -> Data {
+        return Data(
+            bytesNoCopy: UnsafeMutablePointer(mutating: spanNoCopy.__dataUnsafe()),
+            count: spanNoCopy.size(), deallocator: .none)
+    }
+}
```

Comment:
This abstraction is inconsistent with the thesis of UnsafeOverlays. We're returning a reference type (Data) that has a raw pointer in it, with no way to ensure the lifetime of the pointed-to object.

If there are APIs that require Data, and we have a span, probably we need to write overlays for those APIs, where the overlays accept span<T> or similar and then invoke the API directly. The overlay then does the job of encapsulating the unsafe Data object.